packages/mobile/src/components/app-navigator/TopBar.tsx

```diff
@@ -75,7 +76,7 @@ export const TopBar = memo(
     const styles = useStyles()
 
     const { neutralLight4 } = useThemeColors()
-    const navigation = useNavigation()
+    const navigation = useNavigation<MainParamList & SearchParamList>()
```

Comment:
def a TODO for us is reexporting useNavigation with all the types built in

Reply:
Well not all parameters will apply to all navigation contexts, that's why I made it a type parameter with `BaseStackParamList` as the default. That way we can usually avoid having to pass the type parameter but for certain navigation contexts like profile edit or search we can specify that param type

Reply:
oh i see! doesnt navigate have the ability to hit any screen, even if it's outside that stack?

Reply:
I think that's only true if the containing navigator has been rendered: https://reactnavigation.org/docs/nesting-navigators/#navigating-to-a-screen-in-a-nested-navigator

Probably worth discussing synchronously though!